src/watchdog.rs

```diff
@@ -0,0 +1,18 @@
+
+
+
+/// A watchdog timer to reset the processor if software is frozen or stalled.
+pub trait Watchdog {
+    /// Unit of time used by the watchdog
+    type Time;
+    /// Starts the watchdog, typically once this is done the watchdog needs to
+    /// be kicked periodically or the processor is reset. Once started normally
+    /// cannot be stopped
+    fn start(&mut self);
+    /// Set the period of the watchdog 
+    fn set_period<T>(&mut self, period: T) where T: Into<Self::Time>;
```

Comment:
Is it expected that `Self::Time` handles the prevention of invalid periods? Otherwise, `set_period` would have to be fallible, I think.

Additionally, it seems like this might all be more ergonomic if there was no `set_period()`, and instead `start()` took the period as a parameter. With a set-and-forget design like this, I don't see a lot of benefit to separating the act of setting a period into its own operation, but I might be missing something

Reply:
Sure thing, I'll make these changes when I get home from work! With set_period I was considering whether anyone may want to change period after its started. But that may not be supported on everything so I'll change it :+1: